grease/src/Grease/Shape.hs

```diff
@@ -487,17 +493,17 @@ nullPtr =
   let zbyt = TaggedByte{taggedByteTag = NoTag, taggedByteValue = 0}
    in Exactly (take (fromInteger $ CT.intValue ?ptrWidth `div` 8) $ repeat $ zbyt)
 
-constructPtrMemShapeFromRef :: HasPtrWidth w => Subprogram -> MDwarf.TypeRef -> StateT VisitState Maybe (MemShape w NoTag)
-constructPtrMemShapeFromRef sprog ref =
+constructPtrMemShapeFromRef :: HasPtrWidth w => TypeUnrollingBound -> Subprogram -> MDwarf.TypeRef -> StateT VisitState Maybe (MemShape w NoTag)
+constructPtrMemShapeFromRef tyUnrollBound sprog ref =
   do
     mp <- get
     let ct = fromMaybe 0 $ Map.lookup ref mp
     _ <- put (Map.insert ref (ct + 1) mp)
-    if ct > 3
+    if ct > coerce tyUnrollBound
```

Comment:
I think this would be more clear just pattern-matching on `TypeUnrollingBound`, which is also zero-cost.

Reply:
I see that you resolved this - Is it because don't agree? If so, can you say why?

Reply:
i personally wouldnt be a big fan of 
```
case tyUnrollBound of
 TypeUnrollingBound(x) | x > 3 -> ...
 _ -> ...
```

a bit verbose for my taste but lmk

Reply:
What about pattern-matching on it in the argument list? Or adding an accessor `getTypeUnrollingBound` and using it here?

Also, I don't think the `x > 3` would be necessary in either case?

Reply:
err `ct > x` but fine ill match it in the parameter

Reply:
Appreciate your flexibility here. FWIW, `coerce` is fairly uncommon, and it's usually used in circumstances where there would be a performance penalty to using the newtype accessor/pattern match (e.g., `fmap getTypeUnrollingBound someList`).